pkg/workflow/workflowrun/workload.go

```diff
@@ -144,6 +144,22 @@ func (p *WorkloadProcessor) processDelegation() error {
 	}
 
 	p.wfrOper.GetRecorder().Eventf(p.wfr, corev1.EventTypeNormal, "DelegationSucceed", "Delegate stage %s to %s succeeded", p.stg.Name, p.stg.Spec.Delegation.URL)
+
+	// If the task has already been processed by the above RESTful API task, we should not update the status of the task to Waiting
+	latestWfr, err := p.client.CycloneV1alpha1().WorkflowRuns(p.wfr.Namespace).Get(p.wfr.Name, metav1.GetOptions{})
+	if err != nil {
+		log.WithField("wfr", p.wfr.Name).Error("Get latest wfr failed")
+		return err
+	}
+	// The task maybe has been processed by the delegation task and status not needed to be set to Waiting
+	if stg, ok := latestWfr.Status.Stages[p.stg.Name]; ok {
+		if stg.Status.Phase == v1alpha1.StatusSucceeded ||
```

Comment:
I remember that we have defined a method for this status judgement. If not, I think we should define one for this.

Reply:
OK, I will refactor it.